### PR TITLE
ImproveRunningTestsResults

### PR DESCRIPTION
**Pull request by DineshChirnanchu**
# Improve current progress output of dotnet test

#2173

### CODE REVIEW

**Comment by singhsarab** [comment]
@DineshChirnanchu We do not have the spec closure here. Why are we still working on this one ?

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 1]
```diff
@@ -700,5 +706,13 @@ public static void RaiseTestRunWarning(string warningMessage)
 
             Output.Warning(AppendPrefix, warningMessage);
         }
+
+        /// <summary>
+        /// updates the test results
+        /// </summary>
+        public void UpdateTestsStatus()
```
Comment (lauxjpn) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Can directly return the test status as a string. The field is not really used anywhere without first calling `UpdateTestsStatus()`.

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 2]
```diff
@@ -571,7 +576,8 @@ private void TestResultHandler(object sender, TestResultEventArgs e)
                         }
 
                         // Resume the progress indicator after displaying the test result information
-                        this.progressIndicator?.Start();
+                        UpdateTestsStatus();
```
Comment (lauxjpn) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Can be refactored into `this.progressIndicator?.Start(UpdateTestsStatus());`

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 3]
```diff
@@ -547,7 +551,8 @@ private void TestResultHandler(object sender, TestResultEventArgs e)
                             }
 
                             // Resume the progress indicator after displaying the test result information
-                            this.progressIndicator?.Start();
+                            UpdateTestsStatus();
```
Comment (lauxjpn) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Can be refactored into `this.progressIndicator?.Start(UpdateTestsStatus());`

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 4]
```diff
@@ -526,7 +529,8 @@ private void TestResultHandler(object sender, TestResultEventArgs e)
                         DisplayFullInformation(e.Result);
 
                         // Resume the progress indicator after displaying the test result information
-                        this.progressIndicator?.Start();
+                        UpdateTestsStatus();
```
Comment (lauxjpn) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Can be refactored into `this.progressIndicator?.Start(UpdateTestsStatus());`

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 5]
```diff
@@ -505,7 +506,9 @@ private void TestResultHandler(object sender, TestResultEventArgs e)
                         }
 
                         // Resume the progress indicator after displaying the test result information
-                        this.progressIndicator?.Start();
+
+                        UpdateTestsStatus();
```
Comment (lauxjpn) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Can be refactored into `this.progressIndicator?.Start(UpdateTestsStatus());`

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 6]
```diff
@@ -112,6 +112,7 @@ internal enum Verbosity
         private int testsFailed = 0;
         private int testsSkipped = 0;
         private bool testRunHasErrorMessages = false;
+        private string testsStatus = string.Empty;
```
Comment (lauxjpn) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Can be removed.

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 7]
```diff
@@ -700,5 +706,13 @@ public static void RaiseTestRunWarning(string warningMessage)
 
             Output.Warning(AppendPrefix, warningMessage);
         }
+
+        /// <summary>
+        /// updates the test results
+        /// </summary>
+        public void UpdateTestsStatus()
+        {
+            testsStatus = "| Executed :" + testsTotal + "  | Passed :" + testsPassed + "  | Failed: " + testsFailed + "  | Skipped: " + testsSkipped + "";
```
Comment (lauxjpn) [inline src/vstest.console/Internal/ConsoleLogger.cs]: `Executed_:` and `Passed_:` should have no space before the colon and one space after: `Executed:_` and `Passed:_` (replace insert space for "_").

**Review by lauxjpn — COMMENTED** [review]
Just a couple of smaller formatting and refactoring issues.

**Comment by singhsarab** [comment]
This needs more thinking, closing it for now. We'll pick this up later.

**Pull request closed without merging**